docs/development_checklist.md

```diff
@@ -6,8 +6,45 @@ title: "Development Checklist"
 Before you commit any changes, check your work against these requirements:
 
 - All communication to external devices or services must be wrapped in an external Python library hosted on [pypi](https://pypi.python.org/pypi).
-- All dependencies from [pypi](https://pypi.python.org/pypi) are included via the `REQUIREMENTS` variable in your platform or component and [only imported inside functions](creating_component_deps_and_reqs.md) that use them
```

Comment:
We still need this

Reply:
Disregard, I see it later on.

Reply:
The requirements list is still needed in each component. We haven't changed that yet. It will need changing in homeassistant/setup.py.

Reply:
https://github.com/home-assistant/home-assistant/pull/22717

We were a little too fast. I was hoping we'd get this out last night too.